packages/rust/server/tests/helper.rs

```diff
@@ -0,0 +1,144 @@
+use axum_test_helper::{TestClient, TestResponse};
+
+use bubble::models::user::User;
+use bubble::router;
+
+use bubble::routes::group::GroupName;
+use bubble::types::DbPool;
+use sqlx::postgres::PgPoolOptions;
+use sqlx::types::chrono::NaiveDateTime;
+
+use bubble::models::session::Session;
+use std::future::Future;
+use std::{env, thread};
+use tokio::runtime::Runtime;
+use uuid::Uuid;
+
+#[macro_export]
+macro_rules! cleanup {
+    ($structdef: tt, |$dbarg:ident, $resourcesarg: ident| $code: expr) => {{
+        #[derive(Default)]
+        struct CleanupResources
+            $structdef
+
+        let cleanup_resources = CleanupResources::default();
+
+        use sqlx::postgres::PgPoolOptions;
+        use std::env;
+
+        async fn cleanup($resourcesarg: CleanupResources) {
+            let $dbarg = PgPoolOptions::new()
+                .max_connections(5)
+                .connect(&env::var("DB_URL").unwrap())
+                .await
+                .unwrap();
+            $code
+        }
+
+        $crate::helper::Cleanup::new(Box::new(cleanup), cleanup_resources)
+    }};
+
+    (|$dbarg:ident| $code: expr) => {{
+        use sqlx::postgres::PgPoolOptions;
+        use std::env;
+
+        async fn cleanup(_: ()) {
+            let $dbarg = PgPoolOptions::new()
+                .max_connections(5)
+                .connect(&env::var("DB_URL").unwrap())
+                .await
+                .unwrap();
+            $code
+        }
+
+        $crate::helper::Cleanup::new(Box::new(cleanup), ())
+    }}
+}
+
+type CB<F, Resources> = Box<dyn Fn(Resources) -> F>;
+
+pub struct Cleanup<F: 'static + Future + Send, Resources: Default> {
+    cleanup: Option<CB<F, Resources>>,
+    pub resources: Resources,
+}
+
+impl<F: 'static + Future + Send, Resources: Default> Cleanup<F, Resources> {
+    pub fn new(cleanup: CB<F, Resources>, resources: Resources) -> Self {
+        Self {
+            cleanup: Some(cleanup),
+            resources,
+        }
+    }
+}
+impl<F: 'static + Future + Send, Resources: Default> Drop for Cleanup<F, Resources> {
+    fn drop(&mut self) {
+        let cleanup = self.cleanup.take().unwrap();
+        let resources = std::mem::take(&mut self.resources);
+        let future = cleanup(resources);
+        thread::spawn(|| {
+            Runtime::new().unwrap().block_on(future);
+        })
+        .join();
+    }
+}
+
+pub async fn start_server() -> (DbPool, TestClient) {
+    let pool = PgPoolOptions::new()
+        .max_connections(5)
+        .connect(&env::var("DB_URL").unwrap())
+        .await
+        .unwrap();
+
+    let pool2 = pool.clone();
+
+    let router = router::router(pool);
+
+    (pool2, TestClient::new(router))
+}
+
+pub async fn initialize_user(db: &DbPool, _client: &TestClient, username_in: &str) -> (Uuid, User) {
+    let temp_username: String = username_in.parse().unwrap();
```

Comment:
1. .parse().unwrap() is not how to convert &str to String
2. If you need a String, accept a String not a &str